Rework asset store

ClementPasteau opened the pull request:
As part of revamping the Asset Store, the first step is to redo the homepage of the asset store, and suggest to creators packs of assets they can import and use easily.

In this PR:

- Moved all stores stories to their own files with a better structure to test easily
- Fixed a bug where the ExtensionStore would show a loader
- Retrieve the Asset Packs from the backend services
- Show the list on the Asset Store homepage, with a folded filters panel
- Redirect to the previous Asset listing on pack selection, filtered on this pack + open the panel

In the future PRs we will probably:

- Rework the Asset Pack Screen to be independent from the listing, with a better view of the assets it contains + previews + allow adding all of them at once
- Rework the filters to be actual filters (size, color, animations, etc...)
- Rework the Asset screen to be more exhaustive about what the asset contains

Few screenshots in different sizes and themes:

![image](https://user-images.githubusercontent.com/4895034/168093065-1960087c-616a-4433-a402-9c168c2f4862.png)
![image](https://user-images.githubusercontent.com/4895034/168093239-1d8c6cf2-6595-4869-a798-c60ab01be1c7.png)
![image](https://user-images.githubusercontent.com/4895034/168093309-a3eeda9c-fa92-4860-ae9e-fed3d17ce1d9.png)
![image](https://user-images.githubusercontent.com/4895034/168093388-a119183d-6b35-49d3-a961-dc1ff5a8d30e.png)

4ian commented:
Great start! 
Comments in no particular order:
- The focus ring/selection ring is a bit cut for the first line. We might want to do the styling by ourselves. Not a deal breaker
- No feedback when pressing/clicking, let's add something.
- The discover is a Text, but in this design it seems it's used like a title, but it does not look like one. We should at least go for a size="something-that-is-supposed-to-be-a-title", as we can't use the default text styling for something that is not a text. Might be something to check with the design.
- As the filters are not making really a sense for these packs, should we hide them from this first page?

ClementPasteau commented:
> The focus ring/selection ring is a bit cut for the first line. We might want to do the styling by ourselves. Not a deal breaker

Fixed!
> No feedback when pressing/clicking, let's add something.

Fixed!

> The discover is a Text, but in this design it seems it's used like a title, but it does not look like one. We should at least go for a size="something-that-is-supposed-to-be-a-title", as we can't use the default text styling for something that is not a text. Might be something to check with the design.

Fixed!

> As the filters are not making really a sense for these packs, should we hide them from this first page?

For the moment, because those filters are still "Categories", I thought keeping them until the proper filters are in place, and we can then get rid of it.

AlexandreSi commented:
Looks great!

I will have a quick look on how to better display the cards on mobile, the title and the number of assets that are truncated don't look that good.

AlexandreSi commented on newIDE/app/src/AssetStore/AssetsHome.js:
```diff
@@ -0,0 +1,122 @@
+  const classesForGridListItem = useStylesForGridListItem();
+  return (
+    <ThemeConsumer>
+      {muiTheme => (
```
Better use `useContext(ThemeContext)`

AlexandreSi commented on newIDE/app/src/AssetStore/AssetsHome.js:
```diff
@@ -0,0 +1,122 @@
+                        color="primary"
+                        size="body2"
+                      >
+                        {pack.assetsCount} Assets
```
To translate

AlexandreSi commented on newIDE/app/src/AssetStore/index.js:
```diff
@@ -71,13 +81,33 @@ export const AssetStore = ({
+          <Line justifyContent="left" noMargin={isOnHomePage}>
+            {isOnHomePage ? (
+              <Column>
+                <Text size="title">Discover</Text>
```
To translate

AlexandreSi commented on newIDE/app/src/AssetStore/AssetsHome.js:
```diff
@@ -107,7 +112,7 @@ export const AssetsHome = ({
                         size="body2"
                       >
-                        {pack.assetsCount} Assets
+                        {pack.assetsCount} <Trans>Assets</Trans>
```
You can write `<Trans>{pack.assetsCount} Assets</Trans>`. It can give more context to translators

AlexandreSi reviewed (approved):
One last change and it seems ok to me!

4ian commented on newIDE/app/src/AssetStore/index.js:
```diff
@@ -87,41 +120,77 @@ export const AssetStore = ({
+                      allItemsLabel={<Trans>All assets</Trans>}
+                      allFilters={filters}
+                      filtersState={filtersState}
+                      setIsOnHomePage={setIsOnHomePage}
```
setIsOnHomePage seems not a proper name for this prop. Take the point of view of CategoryChooser: 
it does not even know what is a home page. A home page of what?
It just know about its own filters/categories that you give to him. But has no idea when you would want to change if you're on a home page or not - that's not his responsibility! :) 

Instead, what you, as the parent (AssetStore) really want to know is: 
tell me when a tag is clicked and I will react accordingly (here, by... setting we're not on the home page anymore, which is not the business of the child)

Something like:
```suggestion
                      onChoiceChanged={() => setIsOnHomePage(false)}
```

4ian commented on newIDE/app/src/UI/Search/CategoryChooser.js:
```diff
@@ -68,12 +70,15 @@ const MemoizedTagsTree = React.memo<MemoizedTagsTreeProps>(function TagsTree({
       }
       defaultExpanded={[]}
-      onNodeSelect={() => {}}
+      onNodeSelect={() => setIsOnHomePage && setIsOnHomePage(false)}
```
A way to see something is wrong: setIsOnHomePage is called always with false.
This means the prop is not fully used and so is not representing what it's really doing.
It seems here it's just calling it when a label is chosen?

D8H replied:
Yes, I though about it too and kept it like this because it will be removed in a few day.
Do you think it must be done?

4ian replied:
Well I'm always worried of "this is just temporary and I will remove it for sure in two days" 😇
But if you think that's a detail, then no big deal.

D8H replied:
No, you're right it might take some time before the old filters can be removed. I made the change.

4ian commented:
Quick note to make clear a behavior: @D8H, you updated the behavior of the filters/search so that they go automatically to the existing result search page. I think it make sense. Just flagging this for @ClementPasteau.
This is just a first rework so as long as it make sense it's good.
